src/Analyzers/CSharp/CodeFixes/UseUtf8StringLiteral/UseUtf8StringLiteralCodeFixProvider.cs

```diff
@@ -144,7 +140,22 @@ static IArrayCreationOperation FindArrayCreationOperationAncestor(IOperation ope
             }
         }
 
-        private static SyntaxNode CreateArgumentListWithUtf8String(BaseArgumentListSyntax argumentList, Location location, string stringValue)
+        private static string GetUtf8StringValueFromArrayInitializer(IArrayInitializerOperation initializer)
+        {
+            // Get our list of bytes from the array elements
+            using var _ = PooledStringBuilder.GetInstance(out var builder);
+            builder.Capacity = initializer.ElementValues.Length;
+            if (!UseUtf8StringLiteralDiagnosticAnalyzer.TryConvertToUtf8String(builder, initializer.ElementValues))
+            {
+                // We shouldn't get here, because the code fix shouldn't ask for a string value
+                // if the analyzer couldn't convert it
+                throw ExceptionUtilities.Unreachable;
+            }
```

Comment:
```suggestion
            // Can never fail as the analyzer already validated this would work.
            Contract.ThrowIfFalse(UseUtf8StringLiteralDiagnosticAnalyzer.TryConvertToUtf8String(builder, initializer.ElementValues));
```

(note: if the analzyer already validated this, consider just stashing this info on the diagnostic).

Reply:
> consider just stashing this info on the diagnostic

Nah, could be an enourmous string allocation in an diagnostic that is never actioned